### PR TITLE
Nodeagent fornaxcore grpc client

### PR DESCRIPTION
**Pull request by guowenjian90**
use k8s.io.api.core.v1.Node in Node related protobuf messages. Node.Status already defined a lot of status and resources attributes.

implement a client for node agent to get/put message from fornaxcore

### CODE REVIEW

File: pkg/nodeagent/fornaxcore/grpc.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+)
+
+type FornaxCoreConfiguration struct {
+	endpoint       string
+	connTimeout    time.Duration
+	callTimeout    time.Duration
+	maxRecvMsgSize int
+}
+
+type FornaxCore interface {
```
Comment (h-w-chen) [inline pkg/nodeagent/fornaxcore/grpc.go:40]: is this type different from the one auto-generated in fornaxcore.pb.go?

Comment (h-w-chen) [inline pkg/nodeagent/fornaxcore/grpc.go:40]: is this fornax core grpc service client? if it is, can we decorate its name to indicate so (e.g. FornaxCoreNode)?

**Review by h-w-chen — APPROVED** [review]
lgtm; left one minor comment of naming

File: pkg/nodeagent/fornaxcore/grpc.go (hunk 1 above)
Comment (guowenjian90) [reply pkg/nodeagent/fornaxcore/grpc.go]: ack, changed it to FornaxCoreClient

**Pull request merged**